brave/brave-ios
# Fix #7607: Select Multiple Networks in Network Filter Views

## Summary of Changes
- Update to allow selecting multiple networks in the Network Filter Views for Portfolio tab, NFT tab, Activity tab, Edit User Assets, Search Assets, Select Token to Send (in Send view)
- Move test networks to their own section instead of a detail push.
    - This removed a good chunk of UI code. The detail view (test networks), detail rows, view models, etc. no longer needed. 
- Most stores already supported multiple networks in the default flow where 'all networks' was selected. This update piggybacks on that logic.

This pull request fixes #7607

## Submitter Checklist:


## Test Plan:
Test the network filter with All Networks (default) and only a few networks selected in the following views:
1. Portfolio tab
    - Test networks are NOT selected by default (matching existing logic)
5. NFT tab
    - Test networks are NOT selected by default (matching existing logic)
6. Activity tab
    - Test networks selected by default.
7. Edit User Assets
    - Test networks are selected by default.
8. Asset Search (from Portfolio tab)
    - Test networks are selected by default.
9. Select Token to Send (from Send view)
    - Test networks are selected by default.

Show/Hide test networks:
1. Disable 'Show test networks' in Web3 Settings -> Networks
2. Verify test networks are not shown in Network Selection (in Wallet Panel or in Buy/Swap)
3. Verify test networks are not shown in filters (Portfolio / NFT / Activity tab / etc)
4. Enable 'Show test networks' in Web3 Settings -> Networks
5. Verify test networks are shown in Network Selection (in Wallet Panel or in Buy/Swap)
6. Verify test networks are now shown in filters (Portfolio / NFT / Activity tab / etc)

## Screenshots:

Network Filter (Multi-Select) | Network Selection (Single)
--|--
![Network Filter](https://github.com/brave/brave-ios/assets/5314553/42c1d335-463b-495c-9c69-2d521ed61bd0) | ![Network Selection](https://github.com/brave/brave-ios/assets/5314553/8b15d048-4dd1-4452-969a-5fbbf87adf21)


Portfolio tab, NFT tab, Activity tab:

https://github.com/brave/brave-ios/assets/5314553/9985b0a0-cbe0-45d7-9c28-06db6f0649e6

Search Assets, Edit User Assets, Select Token to Send:

https://github.com/brave/brave-ios/assets/5314553/a09bed18-3a2a-4d3c-af21-82a48f8c86c3


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
